657 bot updates for maintenance tasks

jbousquin opened the pull request:
Added maintain.yml and all other file changes are done by bot commits. See issue #657 for links to specific workflow runs and options.

Note: By design this may have merge conflicts because the branch was before a merge that updated reference files to ensure these workflows would bot-update those files.

jbousquin committed:
Create maintain.yaml

Init commit - anticipate it breaking...

jbousquin committed:
Update maintain.yaml

Run on push not pull

jbousquin committed:
Update maintain.yaml

Install via remotes (note this wouldn't be current branch)

jbousquin committed:
Update maintain.yaml

Give it token

github-actions[bot] committed:
Update Notebook

jbousquin committed:
Update maintain.yaml

Edit commit message

jbousquin committed:
Add example data refresh as additional steps in existing workflow (should update 5 rda files)

github-actions[bot] committed:
Update reference files

jbousquin committed:
Try adding to namespace using library() again? Otherwise, and/or may be better, to specify the lib for each function (e.g., EPATADA::TADA_DataRetrieval).

jbousquin committed:
Merge branch '657-bot-updates-for-maintenance-tasks' of https://github.com/USEPA/EPATADA into 657-bot-updates-for-maintenance-tasks

github-actions[bot] committed:
Update example data

jbousquin committed:
Update maintain.yaml

Prep for PR

jbousquin committed:
Run on "develop" not "main

cristinamullin commented on .github/workflows/maintain.yaml:
```diff
@@ -0,0 +1,246 @@
+      - name: Update reference files
+        run: |
+          install.packages('remotes')
+          remotes::install_github("USEPA/TADAShiny", ref = "develop", dependencies = TRUE, force = TRUE)
```
This should install USEPA/EPATADA here not USEPA/TADAShiny

jbousquin committed:
Resolve reviewer comment

Should install EPATADA depends not shiny

jbousquin committed:
Update maintain.yaml

This may or may not address issue of commit not being on a branch

jbousquin commented:
actions/checkout@v4 should be checking out the 'develop' branch, not sure why it is finding itself in a detached head state. Possible because merging is blocked? I'll try being more direct with what to checkout.

jbousquin committed:
ref token for actions/checkout

Add built-in to check out branch as part of checkout

github-actions[bot] committed:
Update reference files

github-actions[bot] committed:
Update example data

jbousquin committed:
Add checks as part of commit

github-actions[bot] committed:
Update example data

request-checks: true

jbousquin committed:
revert maintain.yaml

jbousquin committed:
Update maintain.yaml

Try granting read permissions instead of using token

jbousquin committed:
Update maintain.yaml

Try adding token at a lower level instead of adding it to env

jbousquin committed:
Update maintain.yaml

Not sure how to get token for step not using action. Moved token up to checkout but suspect that will either not make it available to other steps or still skip CI.

jbousquin committed:
Update maintain.yaml

quick fix

jbousquin committed:
Update maintain.yaml

Try setting non-expected env var for use by remotes::install_github

cristinamullin commented on .github/workflows/maintain.yaml:
```diff
@@ -0,0 +1,246 @@
+      - name: Update reference files
+        run: |
+          install.packages('remotes')
+          remotes::install_github("USEPA/EPATADA", auth_token = Sys.getenv("auth_token"), ref = "develop", dependencies = TRUE, force = TRUE)
```
you may need to add library(remotes) after the remotes install here

jbousquin committed:
Revert maintain.yaml

Tried a bunch, going to require a bot token

github-actions[bot] committed:
Update example data

cristinamullin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks

cristinamullin committed:
add library(remotes)

github-actions[bot] committed:
Update example data

jbousquin committed:
Update maintain.yaml

Drop other triggers from this file. Attempt using pre-commit. Low probably of success give I don't think the files are going to be added to the cache and no .pre-commit-config.yaml

jbousquin commented:
Workflow failure confirmed anticipated issues:

1. The updated files are not added to the cache, and
2. No .pre-commit-config.yaml

Will pursue use of other actions next. Note: pre-commit workflow should be easier to figure out with formatter/linting, especially if written in rust ([air](https://www.tidyverse.org/blog/2025/02/air/)).

cristinamullin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks

cristinamullin commented on .github/workflows/maintain.yaml:
```diff
@@ -0,0 +1,230 @@
+          EPATADA:::TADA_UpdateATTAINSOrgIDsRef()
+          EPATADA:::TADA_UpdateATTAINSParamUseOrgRef() # takes a long time
+          EPATADA:::TADA_UpdateATTAINSParameterWQPCharRef()
+          # WQPWQXRefTables.R
```
Do you think we should I create a single maintenance function for use here that includes all of these functions? In addition, I will add the separate function for updating all example data that we discussed.

jbousquin replied:
We could - that would help reduce the size of this file for sure and make it easier to add/remove functions over time.

github-actions[bot] committed:
Update reference files

github-actions[bot] committed:
Update example data

cristinamullin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks

cristinamullin committed:
Merge branch 'maintenance-branch' into 657-bot-updates-for-maintenance-tasks

cristinamullin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks

cristinamullin committed:
use new maintain functions

cristinamullin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks

cristinamullin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks

cristinamullin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks

cristinamullin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks

cristinamullin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks

cristinamullin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks

cristinamullin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks

jbousquin commented:
You can manually trigger [maintenance-scheduled.yaml](https://github.com/USEPA/EPATADA/blob/develop/.github/workflows/maintenance-scheduled.yaml) on your branch/PR to accomplish similar but I think that would commit the change to the maintenance-branch instead. Will play around with/put some thought into that. Also OK with closing this PR if we're happy with just the maintenance bot.

jbousquin committed:
Rename maintain.yaml to maintenance-update.yaml

update to run manually instead of as a pre-commit

jbousquin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks

jbousquin committed:
Update maintenance-scheduled.yaml

Run on schedule only (differentiate from Component File Update)

jbousquin committed:
Merge branch 'develop' into 657-bot-updates-for-maintenance-tasks
